src/Admin/Translation/Controller/MachineTranslationAdminController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+
+namespace App\Admin\Translation\Controller;
+
+use App\Commands\Helpers\CommandHelper;
+use Sonata\AdminBundle\Controller\CRUDController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+abstract class MachineTranslationAdminController extends CRUDController
+{
+  const TYPE_PROJECT = 'TYPE_PROJECT';
+  const TYPE_COMMENT = 'TYPE_COMMENT';
+
+  private string $type;
+
+  public function __construct(string $type)
+  {
+    $this->type = $type;
+  }
+
+  public function listAction(Request $request = null): Response
+  {
+    return $this->renderWithExtraParams('Admin/Translation/admin_machine_translation.html.twig', [
+      'action' => 'list',
+      'trimUrl' => $this->admin->generateUrl('trim')
+    ]);
+  }
+
+  public function trimAction(KernelInterface $kernel): Response
+  {
+    $request = $this->getRequest();
+    $days = $request->get('days');
+
+    if (1 > $days) {
+      return new RedirectResponse($this->admin->generateUrl('list'));
+    }
+
+    $entity = $this->type == self::TYPE_PROJECT ? '--only-project' : '--only-comment';

```

Comment:
Should this maybe be if ($this->type === self::TYPE_PROJECT) {} else if ($this->type === self::TYPE_COMMENT) {} ? If this is ever expanded to include more sections, then this might be missed in the update and cause a bug or something since it really only handles the two cases with this type of statement. Another option would be to create some kind of map or dictionary similar to what we did with the different languages and providers on the UI.